main.go

```diff
@@ -435,19 +437,17 @@ func main() {
 		}()
 	}
 
-	var registerClient = k8s.AdmissionWebhookRegisterClient{
-		Logger: logger.Named("admissionWebhookRegisterClient"),
-	}
+	var source *workloadapi.X509Source
 
-	err = registerClient.Register(ctx, conf)
-	if err != nil {
-		prod.Fatal(err.Error())
+	if strings.EqualFold(conf.WebhookMode, config.ModeAuto) {
+		registerClient := registerClient(ctx, conf, logger)
```

Comment:
`registerClient` as a function name implies that we are registering some client.
I think name `registerSelf` would fit better.
Also, this function could return function `unregister`, so you don't have to deal with `registerClient` in the main function.

Reply:
Thanks, fixed